crates/astria-sequencer/src/bridge/bridge_lock_action.rs

```diff
@@ -145,19 +154,31 @@ impl ActionHandler for BridgeLockAction {
     }
 }
 
-/// returns the length of a serialized `Deposit` message.
-pub(crate) fn get_deposit_byte_len(deposit: &Deposit) -> u128 {
-    use prost::Message as _;
-    let raw = deposit.clone().into_raw();
-    raw.encoded_len() as u128
+/// Returns a modified byte length of the deposit event. Length is calculated with reasonable values
+/// for all fields except `asset` and `destination_chain_address`, ergo it may not be representative
+/// of on-wire length.
+pub(crate) fn calculate_base_deposit_fee(deposit: &Deposit) -> Option<u128> {
+    let variable_length = deposit
+        .asset
+        .to_string()
```

Comment:
Can you extend `Deposit` to provide a function like `display_len()` that calculates this using its formatted representation? Also provide tests in `astria-core` to test that.

We can avoid this allocation.

Reply:
Done in c46fa8a, let me know if this implementation works for you!